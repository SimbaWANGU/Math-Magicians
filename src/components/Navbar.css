* {
  width: 100%;
}

.navBar {
  background-color: rgba(255, 187, 80, 0.5);
  display: flex;
  flex-direction: row;
  height: 60px;
  width: 100%;
  position: relative;
  padding: 0.5rem 0;
  box-shadow: 0 2px 2px 2px rgba(9, 9, 9, 0.23);
}

.websiteTitle {
  font-size: 24px;
  margin: 15px 0 0 10px;
  font-weight: 700;
  color: brown;
}

.navBarList {
  display: flex;
  flex-direction: row;
  margin: auto;
  width: 100%;
  height: 100%;
  padding: 0;
  list-style-type: none;
  text-align: end;
  align-items: center;
  justify-content: center;
}

.navBarList > li {
  padding: 30px 5px 0 0;
}

.navBarList > li > a {
  color: brown;
  font-size: 18px;
  text-decoration: none;
}
