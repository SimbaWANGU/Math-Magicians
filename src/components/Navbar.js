import React from 'react';
import './Navbar.css';

function Navbar() {
  const links = [{
    id: 1,
    path: '/',
    text: 'Home',
  },
  {
    id: 1,
    path: '/calc',
    text: 'Calculator',
  },
  {
    id: 1,
    path: '/quote',
    text: 'Quote',
  }];

  return (
    <nav className="navBar">
      <p className="websiteTitle">Math Magician</p>
      <ul className="navBarList">
        {links.map((link) => <li key={link.id}><a href={link.path}>{link.text}</a></li>)}
      </ul>
    </nav>
  );
}

export default Navbar;
