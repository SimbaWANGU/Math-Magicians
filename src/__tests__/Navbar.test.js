import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';

it('Renders correctly', () => {
  const component = renderer.create(
    <Navbar />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
