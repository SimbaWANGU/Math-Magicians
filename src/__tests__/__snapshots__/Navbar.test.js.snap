// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<nav
  className="navBar"
>
  <p
    className="websiteTitle"
  >
    Math Magician
  </p>
  <ul
    className="navBarList"
  >
    <li>
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/calc"
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        href="/quote"
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
